src/Tribe/Dependency.php

```diff
@@ -175,6 +195,60 @@ public function has_requisite_plugins( $plugins_required = array() ) {
 			return true;
 		}
 
+		/**
+		 * Retrieves Registered Plugin by Class Name from Array
+		 *
+		 * @return array
+		 */
+		public function get_registered_plugin( $class ) {
+			$plugins = $this->registered_plugins;
+
+			return isset( $plugins[ $class ] ) ? $plugins[ $class ] : array();
+		}
+
+		/**
+		 * gets all dependencies or single class requirements
+		 * if parent, co, add does not exist use array as is
+		 * if they do exist check each one in turn
+		 *
+		 */
+		public function dependency_checker( $dependencies = array(), $plugin ) {
+
+			if ( empty( $dependencies ) ) {
+				return true;
+			}
+
+			$failed_dependency = 0;
+
+			$admin_notice  = new Tribe__Admin__Notice__Plugin_Download( $plugin['path'] );
+
+			foreach ( $dependencies as $class => $version ) {
+				$dependency_fail = false;
+
+				// if no class
+				$checked_plugin    = $this->get_registered_plugin( $class );
+				if ( empty( $checked_plugin ) ) {
+					// dependency fail
+					$dependency_fail = true;
+				}
+
+				$is_active = $this->is_plugin_version( $class, $version );
+				if ( empty( $is_active ) ) {
+					// dependency fail
+					$dependency_fail = true;
+				}
+
+				if ( $dependency_fail ) {
```

Comment:
If you invert the two `if` statements and `continue;` in each, we can ditch the `$dependency_fail` variables.